pkg/policy/rule.go

```diff
@@ -447,32 +486,95 @@ func mergeL4Egress(ctx *SearchContext, rule api.EgressRule, ruleLabels labels.La
 	return found, nil
 }
 
+// mergeL4EgressPort merges all rules which share the same port & protocol that
+// select a given set of endpoints. It updates the L4Filter mapped to by the specified
+// port and protocol with the contents of the provided PortRule. If the rule
+// being merged has conflicting L7 rules with those already in the provided
+// L4PolicyMap for the specified port-protocol tuple, it returns an error.
 func mergeL4EgressPort(ctx *SearchContext, endpoints []api.EndpointSelector, r api.PortRule, p api.PortProtocol,
 	proto api.L4Proto, ruleLabels labels.LabelArray, resMap L4PolicyMap) (int, error) {
 
 	key := p.Port + "/" + string(proto)
-	filter, ok := resMap[key]
+	existingFilter, ok := resMap[key]
 	if !ok {
 		resMap[key] = CreateL4EgressFilter(endpoints, r, p, proto, ruleLabels)
 		return 1, nil
 	}
-	l4Filter := CreateL4EgressFilter(endpoints, r, p, proto, ruleLabels)
-	if l4Filter.L7Parser != "" {
-		if filter.L7Parser == "" {
-			filter.L7Parser = l4Filter.L7Parser
-		} else if l4Filter.L7Parser != filter.L7Parser {
-			ctx.PolicyTrace("   Merge conflict: mismatching parsers %s/%s\n", l4Filter.L7Parser, filter.L7Parser)
-			return 0, fmt.Errorf("Cannot merge conflicting L7 parsers (%s/%s)", l4Filter.L7Parser, filter.L7Parser)
+
+	// Create a new L4Filter based off of the arguments provided to this function
```

Comment:
This logic is the same as in mergeL4IngressFilter; this logic should be collapsed together, ideally into a helper function. I can add a TODO for this, but didn't want to spend too much time refactoring due to the short runway to 1.0.

Reply:
This is OK. It was easy to review this way.